File: Nearly Sorted/Nearly Sorted.cpp
Best solution with live coding and explanation - https://youtu.be/Hz7Nz37GkCo
nlogK, k
class Solution {
public:
    void nearlySorted(vector<int>& arr, int k) {
        int n = arr.size();
        priority_queue<int, vector<int>, greater<int>> pq;
        
        // Pushing first k elements in pq
        for (int i = 0; i < k; i++) 
            pq.push(arr[i]);
        
        int i;
        for (i = k; i < n; i++) {
            pq.push(arr[i]);
            arr[i - k] = pq.top();
            pq.pop();
        }
        
        while (!pq.empty()) {
            arr[i - k] = pq.top();
            pq.pop();
            i++;
        }
    }
};

File: Nearly Sorted/Nearly Sorted.java

Best solution with live coding and explanation - https://youtu.be/Hz7Nz37GkCo
nlogK, k

class Solution {
    public void nearlySorted(int[] arr, int k) {
        // code here
        int n = arr.length;
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        
        for(int i=0;i <k; i++){
            pq.add(arr[i]);
        }
        int j;
        for( j=k;j<n;j++){
            pq.add(arr[j]);
            arr[j-k] = pq.poll();
        }
        //
        while(!pq.isEmpty()){
             arr[j-k] = pq.poll();
             j++;
        }
    }
}


